graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/functions/json/JsonUtils.java

```diff
@@ -16,11 +16,116 @@
  */
 package org.graylog.plugins.pipelineprocessor.functions.json;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Joiner;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Maps;
+import org.graylog.autovalue.WithBeanGetter;
+import org.graylog2.jackson.TypeReferences;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
 
 public class JsonUtils {
+    private static final Logger LOG = LoggerFactory.getLogger(JsonUtils.class);
+    private static final String KEY_SEPARATOR = "_";
+    private static final String KEY_VALUE_SEPARATOR = ":";
+    private static final String LIST_SEPARATOR = ",";
+    private static final RemoveNullPredicate<Map.Entry<String, Object>> REMOVE_NULL_PREDICATE = new RemoveNullPredicate<>();
+
+    private JsonUtils() {
+    }
+
+    public static JsonNode extractJson(
+            String value, ObjectMapper mapper, boolean flattenObjects, boolean escapeArrays, boolean deleteArrays)
```

Comment:
It would be nice to have a data class with a builder for the three boolean values. I think we use @AutoValue.Builder for builder classes.